test/puppetlabs/puppetdb/queue_test.clj

```diff
@@ -124,3 +127,42 @@
 
       (is (= foo-cmd-1 (async/<!! c)))
       (is (= foo-cmd-2 (async/<!! c))))))
+
+(deftest test-loading-existing-messages
+  (testing "loading existing messages into a channel"
+    (nio/call-with-temp-dir-path
+     (get-path "target")
+     (str *ns*)
+     (fn [temp-path]
+       (let [q (create-or-open-stockpile temp-path)]
+         (tqueue/store-command q "replace catalog" 1 "foo1" {:message "payload 1"})
+         (tqueue/store-command q "replace catalog" 1 "foo2" {:message "payload 2"})
+         (tqueue/store-command q "replace catalog" 1 "foo3" {:message "payload 3"})
+         (tqueue/store-command q "replace catalog" 1 "foo4" {:message "payload 4"})
+
+         (let [existing-q (create-or-open-stockpile temp-path)
+               command-chan (async/chan 40)]
+           (load-persisted-messages existing-q command-chan)
+           (is (= #{"foo1" "foo2" "foo3" "foo4"}
+                  (set (map :certname (repeatedly 4 #(async/poll! command-chan)))))))))))
```

Comment:
If it's easy, might want to check here that after we read the 4 messages, there's nothing else in the channel.